crypto/ts/quinTree.ts

```diff
@@ -0,0 +1,373 @@
+import type { Leaf, Node, IMerkleProof } from "./types";
+
+/**
+ * An implementation of an incremental Merkle tree
+ * @dev adapted from https://github.com/weijiekoh/optimisedmt
+ */
+export class MACITree {
```

Comment:
I think we call it `IncrementalQuinTree` through the whole project, why not just name it as `IncrementalQuinTree`? will it be conflicted with something else?